tests/ui/type-alias-impl-trait/const_generic_type.infer.stderr

```diff
@@ -6,5 +6,21 @@ LL | async fn test<const N: crate::Bar>() {
    |
    = note: the only supported types are integers, `bool`, and `char`
 
-error: aborting due to 1 previous error
+error: the constant `N` is not of type `u32`
```

Comment:
See the comment:

See the comment:

```
// FIXME: We should reveal the TAITs that end up in where clauses here, otherwise we
// will not be able to match param-env candidates in the old solver, since we don't
// have eq-modulo-normalization. This is less of a problem than it seems, since this
// only matters if we have TAITs in where where clauses, which isn't achievable with
// RPIT anyways.
```